vars/pullGitRepositoryToSapCloudPlatformAbapEnvironment.groovy

```diff
@@ -0,0 +1,185 @@
+            if (finalStatus != 'S') {
+                error "[${STEP_NAME}] Pull Failed"
+            }
+        }
+    }
+}
+
+private String triggerPull(Map configuration, String url, String authToken) {
+
+    String entityUri = null
+    String headerFile = "headerAuth.txt"
+
+    def xCsrfTokenScript = """#!/bin/bash
+        curl -I -X GET ${url} \
+        -H 'Authorization: Basic ${authToken}' \
+        -H 'Accept: application/json' \
+        -H 'x-csrf-token: fetch' \
+        -D ${headerFile} \
+        | awk 'BEGIN {FS=": "}/^x-csrf-token/{print \$2}'
+    """
+
+    def xCsrfToken = sh (
+        script : xCsrfTokenScript,
+        returnStdout: true )
+
+    checkRequestStatus(headerFile)
+
+    String headerFilePost = "headerPost.txt"
+    def scriptPull = """#!/bin/bash
+        curl -X POST \"${url}\" \
+        -H 'Authorization: Basic ${authToken}' \
+        -H 'Accept: application/json' \
+        -H 'Content-Type: application/json' \
+        -H 'x-csrf-token: ${xCsrfToken.trim()}' \
+        --cookie ${headerFile} \
+        -D ${headerFilePost} \
+        -d '{ \"sc_name\": \"${configuration.repositoryName}\" }'
+    """
+    def response = sh (
+        script : scriptPull,
+        returnStdout: true )
+
+    checkRequestStatus(headerFilePost)
+
+    JsonSlurper slurper = new JsonSlurper()
+    Map responseJson = slurper.parseText(response)
+    if (responseJson.d != null) {
+        entityUri = responseJson.d.__metadata.uri.toString()
+        echo "[${STEP_NAME}] Pull Status: ${responseJson.d.status_descr.toString()}"
+    } else {
+        error "[${STEP_NAME}] ${responseJson.error.message.value.toString()}"
+    }
+
+    echo "[${STEP_NAME}] Entity URI: ${entityUri}"
+    return entityUri
+
+}
+
+private String pollPullStatus(String url, String authToken) {
+
+    String headerFile = "headerPoll.txt"
+    String status = "R";
+    while(status == "R") {
+
+        Thread.sleep(5000)
+
+        def pollScript = """#!/bin/bash
+            curl -X GET "${url}" \
+            -H 'Authorization: Basic ${authToken}' \
+            -H 'Accept: application/json' \
+            -D ${headerFile}
+        """
+        def pollResponse = sh (
+            script : pollScript,
+            returnStdout: true )
+
+        checkRequestStatus(headerFile)
+
+        JsonSlurper slurper = new JsonSlurper()
+        Map pollResponseJson = slurper.parseText(pollResponse)
+        if (pollResponseJson.d != null) {
+            status = pollResponseJson.d.status.toString()
+        } else {
+            error "[${STEP_NAME}] ${pollResponseJson.error.message.value.toString()}"
+        }
+        echo "[${STEP_NAME}] Pull Status: ${pollResponseJson.d.status_descr.toString()}"
+    }
+    return status
+}
+
+private void checkRequestStatus(String headerFile) {
+    String headerString = readFile(headerFile)
+    HttpHeader httpHeader = new HttpHeader(headerString)
+    if (httpHeader.statusCode == null) {
```

Comment:
Why are you sure that a request status which is `null` is related to a `503`? In case of a `503`: shouldn't we get a `503` as status code? Might point the troubleshooter into a wrong direction.